Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs

```diff
@@ -0,0 +1,188 @@
+            this.registerAction = registerAction;
+            this.unregisterAction = unregisterAction;
+        }
+
+        /// <summary>
+        /// Initializes the telemetry module.
+        /// </summary>
+        /// <param name="configuration">Telemetry Configuration used for creating TelemetryClient for sending exception statistics to Application Insights.</param>
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            // Core SDK creates 1 instance of a module but calls Initialize multiple times
+            if (!this.isInitialized)
+            {
+                lock (this.lockObject)
+                {
+                    if (!this.isInitialized)
+                    {
+                        this.isInitialized = true;
+
+                        this.telemetryClient = new TelemetryClient(configuration);
+                        this.telemetryClient.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("exstat:");
+
+                        this.metricManager = new MetricManager(this.telemetryClient);
+
+                        this.registerAction(this.CalculateStatistics);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Disposing TaskSchedulerOnUnobservedTaskException instance.
+        /// </summary>
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// IDisposable implementation.
+        /// </summary>
+        /// <param name="disposing">The method has been called directly or indirectly by a user's code.</param>
+        private void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                this.unregisterAction(this.CalculateStatistics);
+                this.metricManager.Dispose();
+            }
+        }
+
+        private void CalculateStatistics(object sender, FirstChanceExceptionEventArgs firstChanceExceptionArgs)
+        {
+            if (executionSyncObject == LOCKED)
+            {
+                return;
+            }
+
+            try
+            {
+                executionSyncObject = LOCKED;
+
+                var exception = firstChanceExceptionArgs?.Exception;
+
+                if (exception == null)
+                {
+                    WindowsServerEventSource.Log.FirstChanceExceptionCallbackExeptionIsNull();
+                    return;
+                }
+
+                WindowsServerEventSource.Log.FirstChanceExceptionCallbackCalled();
+
+                var type = exception.GetType().FullName;
+
+                // obtaining the operation name. At this stage we have no intention to send this telemetry item
+                ExceptionTelemetry fakeTelemetry = new ExceptionTelemetry(exception);
+                this.telemetryClient.Initialize(fakeTelemetry);
+
+                var operation = fakeTelemetry.Context.Operation.Name;
```

Comment:
Can we instead get that operation from the Request Telemetry in HttpContext via our extension of HttpContext? Then, if it's not available (not classic ASP.NET) - check for operation in the storage we have for OperationsAPI? If still nothing, do not assign anything to operation... Or do we expect that there is a third way to set the operation and this fakeTelemetry item will help us to get to that better?